tests/app/utilities/test_decimal_places.py

```diff
@@ -0,0 +1,129 @@
+        (Decimal("123.4"), "es_US", "123.4"),
+        (Decimal("123.40"), "en_GB", "123.40"),
+        (Decimal("123.400"), "en_GB", "123.400"),
+        (Decimal("123.4000"), "en_GB", "123.4000"),
+        (Decimal("123.4000"), "pl_PL", "123,4000"),
+        (Decimal("123.40000"), "en_GB", "123.40000"),
+        (Decimal("123.40000"), "ja_JP", "123.40000"),
+        (Decimal("123434.7678"), "en_GB", "123,434.7678"),
+        (Decimal("123434.7678"), "hu_HU", "123 434,7678"),
+        (Decimal("123.45678"), "en_GB", "123.45678"),
+        (Decimal("2344.6533"), "en_GB", "2,344.6533"),
+        (1000, "en_GB", "1,000"),
+        (1000, "en_US", "1,000"),
+        (10000, "en_GB", "10,000"),
+        (10000, "es_ES", "10.000"),
+        (100000000, "en_GB", "100,000,000"),
+        (0, "en_GB", "0"),
+        (0, "de_DE", "0"),
+        (Decimal("0.00"), "en_GB", "0.00"),
+        (Decimal("0.000"), "en_GB", "0.000"),
+        (Decimal("0.000"), "es_ES", "0,000"),
+        (Decimal("0.00000"), "en_GB", "0.00000"),
+        (Decimal("0.00000"), "es_ES", "0,00000"),
+    ),
+)
+def test_custom_format_decimal(value, locale_string, expected_result):
+    result = custom_format_decimal(value, locale_string)
+
+    assert unicodedata.normalize("NFKD", result) == expected_result
+
+
+@pytest.mark.parametrize(
+    "value, measurement_unit, locale_string, length, expected_result",
+    (
+        (123, "mile", "en_GB", "short", "123 mi"),
+        (
+            Decimal("0.123"),
+            "millimeter",
+            "en_GB",
+            "short",
+            "0.123 mm",
+        ),
+        (
+            Decimal("0.123"),
+            "millimeter",
+            "es_ES",
+            "short",
+            "0,123 mm",
+        ),
+        (123, "centimeter", "en_GB", "short", "123 cm"),
+        (123, "centimeter", "pl_PL", "short", "123 cm"),
+        (123, "kilometer", "en_GB", "long", "123 kilometres"),
+        (3, "kilometer", "en_GB", "long", "3 kilometres"),
+        (3, "kilometer", "hu_HU", "long", "3 kilométer"),
+        (Decimal("1.2"), "kilometer", "en_GB", "long", "1.2 kilometres"),
+        (Decimal("1.20"), "kilometer", "en_GB", "long", "1.20 kilometres"),
+        (Decimal("1.200"), "kilometer", "en_GB", "long", "1.200 kilometres"),
+        (Decimal("1.2000"), "kilometer", "en_GB", "long", "1.2000 kilometres"),
+        (Decimal("1.20000"), "kilometer", "en_GB", "long", "1.20000 kilometres"),
+        (Decimal("1.20000"), "kilometer", "de_DE", "long", "1,20000 Kilometer"),
+        (Decimal("1.2"), "kilometer", "pl_PL", "long", "1,2 kilometra"),
+        (Decimal("1.2345"), "kilometer", "en_GB", "long", "1.2345 kilometres"),
+        (Decimal("1.2345"), "kilometer", "es_ES", "long", "1,2345 kilómetros"),
+        (123, "mile", "en_GB", "short", "123 mi"),
+        (123, "mile", "en_GB", "narrow", "123mi"),
+        (123, "mile", "en_US", "narrow", "123mi"),
+    ),
+)
+def test_custom_format_unit(
```

Comment:
Same as above, why do we have this and `TEST_FORMAT_UNIT_PARAMS`. If you handle the locale you have a single test set for both?.

Reply:
Updated 👍